### ISSUE TITLE
transients_ columns should only appear in metadata table if simulating transients

### ISSUE THREAD

**Issue by moustakas** [desihub/desisim#529]
Spun off from https://github.com/desihub/desisim/pull/525 but also impacting `desitarget` in https://github.com/desihub/desitarget/issues/603.

**Comment by sybenzvi** [comment]
@moustakas, I'm finally looking into this.

My intention was always to conditionally insert the TRANSIENT_ columns into the truth tables when the user adds transients to galaxy templates through desisim. The default behavior should be to exclude the columns. I set up a boolean in desisim to manage it but found its state was being lost when the generation code was called in desitarget.

I know where the problem is, but unfortunately when writing the code I couldn't figure out an easy fix, so I left the columns in place with the intention of revisiting the issue. Then I got distracted and my "temporary" non-solution was merged into the master branch.

**Comment by moustakas** [comment]
@sybenzvi, absolutely! I just opened this ticket so we wouldn't lose track. When I find some cycles I can also help figure out the right desitarget<-->desisim hooks. For example, it'd be great to add a transients simulation option to `select_mock_targets`, which should be straightfoward.
